proto/substrait/algebra.proto

```diff
@@ -654,6 +656,9 @@ message HashJoinRel {
     JOIN_TYPE_RIGHT_SEMI = 6;
     JOIN_TYPE_LEFT_ANTI = 7;
     JOIN_TYPE_RIGHT_ANTI = 8;
+    // This join is useful for nested sub-queries where we need exactly one record in output (or throw exception)
+    // See Section 3.2 of https://15721.courses.cs.cmu.edu/spring2018/papers/16-optimizer2/hyperjoins-btw2017.pdf
```

Comment:
Can we just remove these comments from all the proto files?  If we want to further justify the single join (which might make sense as it is not ubiquitous yet) then perhaps we put the link to section 3.2 in the site docs where the single join operator is described?

Reply:
Done.